lib/utils/BaseReporter.js

```diff
@@ -63,45 +66,76 @@ class BaseReporter extends events.EventEmitter {
         })
 
         this.on('runner:start', (runner) => {
-            if (!this.stats.runner[runner.cid]) {
-                this.stats.runner[runner.cid] = {
-                    start: new Date(),
-                    capabilities: runner.capabilities,
-                    config: runner.config,
-                    tests: []
-                }
-            }
+            this.getSpecStats(runner)
```

Comment:
Hmm. I understand this would make it simpler to test (presumably you would mock `this.getSpecStats`?), but can't see a way to do it and still keep the abstraction contained within `this.getSpecStats`. Otherwise, for example, `runner:command`, `runner:result`, `test:start`, etc. would all need to assign the result from `this.getTestStats()` to the following nested structure `this.stats.runner[test.cid].specs[test.specHash].suites[test.parent].tests[test.title]`. Is that what you mean?